### ISSUE TITLE
CLI downloading a "big" file.

### ISSUE THREAD

**Issue by FrozenSource** [yhirose/cpp-httplib#659]
Hello I am trying to download a "big" zipped file from a server side also httplib.
`httplib::Client cli("http://localhost:8080");
    string sBuffer;
    auto res = cli.Get("/package/dl.zip", [&](const char *data, size_t data_length) {
        sBuffer.append(data, data_length);
        return true;
    });`
But assuming that it will only perform one http request with the http buffer. is there a way to download a whole file using this library or am I missing something?

**Comment by FrozenSource** [comment]
![image](https://user-images.githubusercontent.com/14978036/93671157-e9a77a80-faa0-11ea-918b-6296e120e8d7.png)
Somehow I am missing 2kb

**Comment by yhirose** [comment]
@FrozenSource, thanks for your feedback. I actually don't know why, but I think the gzip related unit tests can help you. Assigning appropriate HTTP headers for you environment are important. Hope it helps! 

https://github.com/yhirose/cpp-httplib/blob/b8cf739d275d636fa3a30f56e16c26080ef86578/test/test.cc#L2455-L2586

**Comment by FrozenSource** [comment]
Thanks I did indeed try this. However it did not make any differences. Through Chrome it downloads the file correctly so it should not have to do anything with the server side of httplib but rather the client. I tried copying all request headers too however this was just an Accept header and did not change anything. Thus I started doing a hex compare using flexhex. And there are some wierd shifts and these are all through the file causing the corruption of the zip. 
![image](https://user-images.githubusercontent.com/14978036/93687375-bde6bd80-fabd-11ea-8491-493f632ce1d5.png)
